test/tests/functional/pbs_test_tpp.py

```diff
@@ -386,30 +429,414 @@ def test_multiple_comm_with_mom(self):
         Node 2 : Mom
         Node 3 : Comm
         """
-        if self.moms.values()[0].shortname == self.server.shortname:
-            self.momA = self.moms.values()[0]
-            self.momB = self.moms.values()[1]
-        else:
-            self.momA = self.moms.values()[1]
-            self.momB = self.moms.values()[0]
-        if self.comms.values()[0].shortname == self.server.shortname:
-            self.comm2 = self.comms.values()[1]
-        else:
-            self.comm2 = self.comms.values()[0]
-        self.hostA = self.momA.shortname
-        self.hostB = self.momB.shortname
-        self.hostC = self.comm2.shortname
-        nodes = [self.hostA, self.hostB, self.hostC]
-        self.node_list.extend(nodes)
+        self.common_setup()
         a = {'PBS_COMM_ROUTERS': self.hostA}
         self.set_pbs_conf(host_name=self.hostC, conf_param=a)
         b = {'PBS_LEAF_ROUTERS': self.hostC}
         self.set_pbs_conf(host_name=self.hostB, conf_param=b)
         self.common_steps(job=True, interactive=True, resv=True,
                           resv_job=True)
 
+    def common_steps_for_comm_failover(self):
+        """
+        This function has common steps for comm failover used in
+        diff tests
+        """
+        self.common_steps(job=True, interactive=True)
+        rid = self.submit_resv()
+        jid = self.submit_job(rid=rid, resv_job=True, sleep=60)
+        resv_exp_attrib = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, resv_exp_attrib, rid, offset=10)
+        job_exp_attrib = {'job_state': 'R'}
+        self.server.expect(JOB, job_exp_attrib, id=jid)
+        self.comm2.stop('-KILL')
+        for mom in self.moms.values():
+            self.server.expect(NODE, {'state': 'free'}, id=mom.shortname)
+        self.server.expect(RESV, resv_exp_attrib, rid)
+        self.server.expect(JOB, job_exp_attrib, id=jid)
+        self.comm2.start()
+        self.comm.stop('-KILL')
+        for mom in self.moms.values():
+            self.server.expect(NODE, {'state': 'free'}, id=mom.shortname)
+        self.server.expect(RESV, resv_exp_attrib, rid)
+        self.server.expect(JOB, job_exp_attrib, id=jid)
+
+    @requirements(num_moms=2, num_comms=2)
+    def test_comm_failover_with_multiple_PBS_LEAF_ROUTERS(self):
```

Comment:
I suggest to name this test as just "test_comm_failover" to avoid confusion; since comm failover itself indicates presence of multiple values in PBS_LEAF_ROUTERS variable of pbs.conf